aodndata/aatams_sattag_qc_dm/aatams_sattag_qc_dm_schema.py

```diff
@@ -0,0 +1,685 @@
+        self.zip_schema = FILENAMES_IN_ZIP_SCHEMA
+
+        self.file_schemas = {
+            "metadata": METADATA_SCHEMA,
+            "ctd": CTD_SCHEMA,
+            "diag": DIAG_SCHEMA,
+            "dive": DIVE_SCHEMA,
+            "haulout": HAULOUT_SCHEMA,
+            "ssmoutputs": SSMOUTPUTS_SCHEMA,
+            "summary": SUMMARY_SCHEMA,
+        }
+
+        self.same_keys_content = []
+        self.cross_content_scope = {
+            "cid": "sattag_program",
+        }
+
+    def validate_zip_names(self, file_list):
+        """Validate the file names.
+
+        Args:
+          file_list: a list of file names.
+
+        Returns:
+          Any: The result of schema call.
+
+        Raises:
+          SchemaError: If the zip files are non-conformant.
+
+        """
+        self.report("Validating filename conventions")
+        return Schema(self.zip_schema).validate(file_list)
+
+    def validate_headers_only(self, file_list):
+        """Validate the headers of a list of csv files.
+
+        Args:
+          file_list: a list of csv files.
+
+        Returns:
+          Dict[str,Any]: A dictionary with the result of the schema call for every file header.
+
+        Raises:
+          SchemaError: if any header is non-conformant.
+
+        """
+        headers = {}
+        for file in file_list:
+            schema_name = self.file2schema(file)
+            schema = self.file_schemas[schema_name]
+            self.report("Loading headers in %s" % file)
+            header = self.load_csv_header(file)
+            self.report("Validating headers in %s" % file)
+            headers[schema_name] = self.validate_header(header, schema)
+        return headers
+
+    def quick_validation(self, file_list):
+        """Perform validation only in zip file names and headers.
+
+        Args:
+          file_list: a list of csv files.
+
+        Returns:
+          None: if all valid
+
+        Raises:
+          SchemaError: if zip or headers are non-conformant.
+
+        """
+        self.validate_zip_names(file_list)
+        self.validate_headers_only(file_list)
+
+    pass
+
+    def extensive_validation(self, file_list):
+        """Extensive validation of files within a AATAMS_DM zip.
+
+        1. Filename conventions
+        2. Metadata headers
+        3. Data rows types
+        4. Data rows relationships
+        5. Data rows cross relationships with the metadata
+
+        Args:
+          file_list: a list of csv files.
+
+        Returns:
+          None: if all files are conformant.
+
+        Raises:
+          SchemaError: if zip, headers, types and content are non-conformant.
+
+        """
+        self.validate_zip_names(file_list)
+        schema_name = "metadata"
+        metadata_file = [x for x in file_list if self.file2schema(x) == schema_name]
+
+        self.headers = {}
+        self.headers["metadata"], self.metadata = self.validate_file(metadata_file[0])
+
+        if self.cross_content_scope:
```

Comment:
why creating this condition since this is always defined as "cid": "sattag_program" anyway ?

Reply:
yes - the cross validation was broken when I incompletely rebase - fixed the logic in new commits